src/Orleans.Transactions/Abstractions/TransactionInfo.cs

```diff
@@ -86,33 +85,57 @@ public void Join(ITransactionInfo x)
             this.joined.Enqueue((TransactionInfo)x);
         }
 
+        public OrleansTransactionAbortedException MustAbort(SerializationManager sm)
+        {
+
+            if (OriginalException != null)
+            {
+                var reader = new BinaryTokenStreamReader(OriginalException);
+                return sm.Deserialize<OrleansTransactionAbortedException>(reader);
+            }
```

Comment:
Why serialize this yourself?  Doesn't codegen do this for you if you have a field "Exception OriginalException;"?

Reply:
You would think so... I tried that first. But got weird failure from Orleans serializer. What you see here represents a workaround.

Reply:
When issues like this pop up, please feel free to message in Gitter/email/IM - we can help